packages/core/src/internal/utils/responsive.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2016-2020 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import { LitElement } from 'lit-element';
+
+export type ResponsiveComponent = LitElement & { layout: string; responsive: boolean; layoutStable: boolean };
+
+export interface LayoutConfig {
+  layouts: string[];
+  initialLayout: string;
+}
+
+export function elementResize(element: HTMLElement, callbackFn: () => void) {
+  const observer = new ResizeObserver(() => callbackFn());
+  observer.observe(element);
+  (observer as any).__testTrigger = callbackFn; // hook to trigger resize event as ResizeObserver does not run in headless chrome.
+  return observer;
+}
+
+/**
+ * Given a ResponsiveComponent this function will loop through a list of layout
+ * options and change the layout of the component until the components layout
+ * condition is satisfied.
+ */
+export function updateComponentLayout(component: ResponsiveComponent, layoutConfig: LayoutConfig, fn: () => void) {
```

Comment:
This is a generic function that can be used to trigger resize layout changes. Form inputs use this to determine if the layout is broken and if it should shift to a smaller more compact layout.